common/cmake/devices.cmake

```diff
@@ -12,6 +12,7 @@ function(DEFINE_ARCH)
   # ~~~
   # DEFINE_ARCH(
   #    ARCH <arch>
+  #    PART <part>
```

Comment:
I don't agree that PART is part of ARCH?

Reply:
So, the main issue is that right now, most of the scripts do require the PART.
Right now, the PART is passed only when creating the board. The reason I assigned the PART to DEFINE_ARCH is that, for instance, artix 50T and 200T are actually different architectures with different structures (even though they share the db info).

This is untrue for the 35T which do actually share tilegrid and tileconn. My plan is to define only one arch for each one of these families, that will share rr_graph, arch.xml and all that is equal. While the pins and can be retrieved in the corresponding location in the db

Reply:
So as a concrete example, there are 4 pinouts for the 200T, but likely only 1 RR graph.  It does not make sense to generate 4 lookaheads for the same graph, as this is very expensive.

Maybe `project_xray_prepare_database` should have the following arguments:

`PARTS <list of parts>`

and then the database generation picks a part when it needs too.  This also enables verification that each graph is in fact the same, as an optional follow on step.

In additional maybe an argument like
`DEFAULT_PART` or `EXAMPLE_PART`, where the documentation lists this part as the prototypical part for the family of parts with the same graph.

Reply:
With the `list of parts` you mean parts independent one to another (50T and 200T) or parts that are of the same "parts family"?

In addition I need to figure out how to pass the PART that needs to be fed as argument to the various scripts.

Reply:
list of parts would be all parts with the same RR graph (e.g. all 50T).

> In addition I need to figure out how to pass the PART that needs to be fed as argument to the various scripts.

So I think there are two cases.  When generating the connection database, the `PROTOTYPICAL_PART` is sufficient.  For things like `fasm2frames`, IO placement, and `fasm2bels`, you probably want the part defined in `add_xc7_device`.

Reply:
I have adopted a slightly different approach.
Given that, for instance, the tile_import scripts require the PART, I have passed to `define_arch` a prototypical PART, which is common to each part of the family (all the artix7).

For the `prepare database`, instead, a list of sub-families are passed as a list (50T sub-family, 200T sub-family) and the generic channels.db is created for each of them.

This way we can define one device for each of these families, and a single rr_graph (and company) are going to be produced for that sub-family only.

When defining the boards, instead, we can use the available devices and they will use the correct package_pins files for the `after FASM` scripts.